### PR TITLE
DB-10084 HBase flush and compactions are aware of min retention period 

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 1]
```diff
@@ -1666,6 +1669,11 @@ public void addMinRetentionPeriodColumn(TransactionController tc) throws Standar
             DataTypeDescriptor dtd = DataTypeDescriptor.getBuiltInDataTypeDescriptor(Types.BIGINT, 1);
             tc.addColumnToConglomerate(td.getHeapConglomerateId(), colNumber, storableDV, dtd.getCollationType());
 
+            try {
```
Comment (ipraznik-splice) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java:1672]: 10 sec delay has to be removed

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java [hunk 2]
```diff
@@ -399,11 +399,11 @@ public void setPurgeDeletedRows(boolean purgeDeletedRows) {
         this.purgeDeletedRows = purgeDeletedRows;
     }
 
-    public Long getMinRetainedVersions() {
+    public Long minRetainedPeriod() {
```
Comment (ipraznik-splice) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java:402]: should it be in Java style getMinRetainedPeriod?

**Review by ipraznik-splice — CHANGES_REQUESTED** [review]

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 3]
```diff
@@ -1666,6 +1669,11 @@ public void addMinRetentionPeriodColumn(TransactionController tc) throws Standar
...
+                Thread.sleep(10000);
```
Comment (ipraznik-splice) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java:1673]: 10 sec delay has to be removed

**Review by ipraznik-splice — CHANGES_REQUESTED** [review]
please remove 10 sec delay as it was already removed in previous commits

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java (hunk 2 above)
Comment (arnaud-splice) [reply db-engine/src/main/java/com/splicemachine/db/iapi/sql/dictionary/TableDescriptor.java]: Done

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java (hunk 3 above)
Comment (arnaud-splice) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java]: Done

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1584/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1594/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1598/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1604/

**Comment by ascend1** [comment]
Are changes made by Youssef already got reviewed?

**Comment by arnaud-splice** [comment]
> Are changes made by Youssef already got reviewed?

Yes :)

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1613/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1619/

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1635/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by ipraznik-splice — APPROVED** [review]

**Review by ascend1 — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
jenkins please test branch @hdp6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**